wikimedia/wikipedia-ios
# Fix Core Data value transformer warnings

**Fixes Phabricator ticket:** 
https://phabricator.wikimedia.org/T231987

### Notes
Creates a custom `WMFSecureUnarchiveFromDataTransformer` that allows `MTLModel` and the other objects we transform. Makes our `MTLModel` subclasses conform to `NSSecureCoding`.  Utilizes `WMFSecureUnarchiveFromDataTransformer` for all transformable Core Data properties.

I did some additional cleanup of the `WMFContentGroup` model to remove a deprecated class. I also fixed an error that surfaced with the `NSKeyedUnarchiver`s that changed when dropping iOS 11.

### Test Steps
1. Use the app normally with a prior version
2. Run the app with this branch
3. Ensure the feed data, user preferences, and event logging continue to work as normal
